globus_cli/services/transfer/task/cancel.py

```diff
@@ -10,16 +11,57 @@
 @click.command('cancel', short_help='Cancel a Task',
                help='Cancel a Task owned by the current user')
 @common_options
-@task_id_option(helptext='ID of the Task which you want to cancel')
-def cancel_task(task_id):
+@task_id_option(helptext='ID of the Task which you want to cancel',
+                required=False)
+@click.option('--all', '-a', is_flag=True,
+              help='Cancel all in-progress tasks that you own')
+def cancel_task(all, task_id):
     """
     Executor for `globus transfer task cancel`
     """
+
+    if bool(all) + bool(task_id) != 1:
+        raise click.UsageError('You must pass EITHER the special --all flag '
+                               'to cancel all in-progress tasks OR a single '
+                               'task ID to cancel.')
+
     client = get_client()
 
-    res = client.cancel_task(task_id)
+    if all:
+        from sys import maxsize
+        task_ids = [
+            task_row['task_id']
+            for task_row in client.task_list(
+                filter='type:TRANSFER,DELETE/status:ACTIVE,INACTIVE',
+                fields='task_id',
+                num_results=maxsize,  # FIXME want to ask for "unlimited" set
+            )
+        ]
+
+        if not task_ids:
+            raise click.ClickException('You have no in-progress tasks.')
```

Comment:
Is this the right behavior?
I think this should just be a successful exit.

It's vacuous, but the user is in the "good" state of having all active tasks canceled.